### PR TITLE
Replace manual get/set code in GlobalAppSettings with GETSET_PROPERTY

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
## Summary of the Pull Request

This pull request removes all of the custom `Get` and `Set` implementations from GlobalAppSettings and replaces them with `GETSET_PROPERTY`. This will be required if we ever convert it to a WinRT class, but for now it's simply niceness-improving.

## References

Required #5847 to land.

## PR Checklist
* [ ] Closes norhing
* [x] CLAd
* [x] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already

### CODE REVIEW

File: src/cascadia/TerminalApp/GlobalAppSettings.h [hunk 1]
```diff
@@ -27,6 +27,11 @@ namespace TerminalAppLocalTests
 namespace TerminalApp
 {
     class GlobalAppSettings;
+
+    struct LaunchPosition
+    {
+        std::optional<int> x, y;
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/GlobalAppSettings.h:33]: the tiniest nit: I'd put these on separate lines

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp [hunk 2]
```diff
@@ -250,14 +115,14 @@ std::optional<int32_t> GlobalAppSettings::GetInitialY() const noexcept
...
     settings.KeyBindings(GetKeybindings());
-    settings.InitialRows(_initialRows);
-    settings.InitialCols(_initialCols);
-    settings.RowsToScroll(_rowsToScroll);
-
-    settings.WordDelimiters(_wordDelimiters);
-    settings.CopyOnSelect(_copyOnSelect);
-    settings.ForceFullRepaintRendering(_forceFullRepaintRendering);
-    settings.SoftwareRendering(_softwareRendering);
+    settings.InitialRows(_InitialRows);
```
Comment (miniksa) [inline src/cascadia/TerminalApp/GlobalAppSettings.cpp:118]: Something something "underscore with capital letter is supposed to be reserved."

Comment (DHowett) [reply src/cascadia/TerminalApp/GlobalAppSettings.cpp:118]: Meh that's what our macros do and it's a giant annoying but simply mechanical change to fix it.

Comment (miniksa) [reply src/cascadia/TerminalApp/GlobalAppSettings.cpp:118]: I don't care that much. I just thought we had some slow semblance of trying to not introduce more of them. But if it's a pain in the ass, just forget it. It's not worth it and no one abides by it in practice.

**Comment by DHowett** [comment]
> stamp those out

I want to stamp them out, but it would blow up this PR more than a little bit -- every cpp file subjugate to a header where we use GETSET_PROPERTY will change.

**Pull request merged**